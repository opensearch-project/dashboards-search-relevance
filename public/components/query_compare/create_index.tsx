/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiPageBody, EuiEmptyPrompt, EuiLink } from '@elastic/eui';

import { Header } from '../common/header';
import { CoreStart } from '../../../../../src/core/public';

interface CreateIndexProps {
  chrome: CoreStart['chrome'];
  application: CoreStart['application'];
}

export const CreateIndex = ({ chrome, application }: CreateIndexProps) => {
  const navGroupEnabled = chrome.navGroup.getNavGroupEnabled();
  return (
    <>
      <Header />
      <EuiPageBody>
        <EuiEmptyPrompt
          title={<h2>Create an index to start comparing search results. </h2>}
          body={
            <p>
              Before you can query data, you need to index it.{' '}
              <EuiLink
                href="https://opensearch.org/docs/latest/opensearch/index-data/"
                target="_blank"
              >
                Learn how to index your data
              </EuiLink>
              , or{' '}
              <EuiLink
                {...(navGroupEnabled
                  ? {
                      onClick: () => {
                        application.navigateToApp('import_sample_data');
                      },
                    }
                  : { href: '/app/home#/tutorial_directory' })}
              >
                add sample data
              </EuiLink>{' '}
              to OpenSearch Dashboards.
            </p>
          }
        />
      </EuiPageBody>
    </>
  );
};
