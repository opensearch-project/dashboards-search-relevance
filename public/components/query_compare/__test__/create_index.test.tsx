/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { CreateIndex } from '../create_index';

describe('Create index component', () => {
  configure({ adapter: new Adapter() });

  it('Renders create index component', async () => {
    const wrapper = mount(<CreateIndex />);

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
