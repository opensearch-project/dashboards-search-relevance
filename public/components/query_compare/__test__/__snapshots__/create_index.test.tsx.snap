// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create index component Renders create index component 1`] = `
<CreateIndex>
  <Header>
    <EuiPanel
      borderRadius="none"
      grow={false}
      hasBorder={false}
      hasShadow={false}
      style={
        Object {
          "borderBottom": "1px solid #D3DAE6",
        }
      }
    >
      <div
        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero"
        style={
          Object {
            "borderBottom": "1px solid #D3DAE6",
          }
        }
      >
        <EuiTitle>
          <h1
            className="euiTitle euiTitle--medium"
          >
            Compare search results
          </h1>
        </EuiTitle>
        <EuiText>
          <div
            className="euiText euiText--medium"
          >
            <p>
              Compare results using the same search text with different queries.
               
              <EuiLink
                href="https://opensearch.org/docs/latest/search-plugins/search-relevance"
                target="_blank"
              >
                <a
                  className="euiLink euiLink--primary"
                  href="https://opensearch.org/docs/latest/search-plugins/search-relevance"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Learn more
                  <EuiIcon
                    aria-label="External link"
                    className="euiLink__externalIcon"
                    size="s"
                    type="popout"
                  >
                    <EuiIconEmpty
                      aria-hidden={true}
                      aria-label="External link"
                      className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                      focusable="false"
                      role="img"
                      style={null}
                    >
                      <svg
                        aria-hidden={true}
                        aria-label="External link"
                        className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      />
                    </EuiIconEmpty>
                  </EuiIcon>
                  <EuiScreenReaderOnly>
                    <span
                      className="euiScreenReaderOnly"
                    >
                      <EuiI18n
                        default="(opens in a new tab or window)"
                        token="euiLink.newTarget.screenReaderOnlyText"
                      >
                        (opens in a new tab or window)
                      </EuiI18n>
                    </span>
                  </EuiScreenReaderOnly>
                </a>
              </EuiLink>
            </p>
          </div>
        </EuiText>
      </div>
    </EuiPanel>
  </Header>
  <EuiPageBody>
    <div
      className="euiPageBody euiPageBody--borderRadiusNone"
    >
      <EuiEmptyPrompt
        body={
          <p>
            Before you can query data, you need to index it.
             
            <EuiLink
              href="https://opensearch.org/docs/latest/opensearch/index-data/"
              target="_blank"
            >
              Learn how to index your data
            </EuiLink>
            , or 
            <EuiLink
              href="/app/home#/tutorial_directory"
            >
              add sample data 
            </EuiLink>
            to OpenSearch Dashboards.
          </p>
        }
        title={
          <h2>
            Create an index to start comparing search results. 
          </h2>
        }
      >
        <div
          className="euiEmptyPrompt"
        >
          <EuiTitle
            size="m"
          >
            <h2
              className="euiTitle euiTitle--medium"
            >
              Create an index to start comparing search results. 
            </h2>
          </EuiTitle>
          <EuiTextColor
            color="subdued"
          >
            <span
              className="euiTextColor euiTextColor--subdued"
            >
              <EuiSpacer
                size="m"
              >
                <div
                  className="euiSpacer euiSpacer--m"
                />
              </EuiSpacer>
              <EuiText>
                <div
                  className="euiText euiText--medium"
                >
                  <p>
                    Before you can query data, you need to index it.
                     
                    <EuiLink
                      href="https://opensearch.org/docs/latest/opensearch/index-data/"
                      target="_blank"
                    >
                      <a
                        className="euiLink euiLink--primary"
                        href="https://opensearch.org/docs/latest/opensearch/index-data/"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        Learn how to index your data
                        <EuiIcon
                          aria-label="External link"
                          className="euiLink__externalIcon"
                          size="s"
                          type="popout"
                        >
                          <EuiIconEmpty
                            aria-hidden={true}
                            aria-label="External link"
                            className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                            focusable="false"
                            role="img"
                            style={null}
                          >
                            <svg
                              aria-hidden={true}
                              aria-label="External link"
                              className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                              focusable="false"
                              height={16}
                              role="img"
                              style={null}
                              viewBox="0 0 16 16"
                              width={16}
                              xmlns="http://www.w3.org/2000/svg"
                            />
                          </EuiIconEmpty>
                        </EuiIcon>
                        <EuiScreenReaderOnly>
                          <span
                            className="euiScreenReaderOnly"
                          >
                            <EuiI18n
                              default="(opens in a new tab or window)"
                              token="euiLink.newTarget.screenReaderOnlyText"
                            >
                              (opens in a new tab or window)
                            </EuiI18n>
                          </span>
                        </EuiScreenReaderOnly>
                      </a>
                    </EuiLink>
                    , or 
                    <EuiLink
                      href="/app/home#/tutorial_directory"
                    >
                      <a
                        className="euiLink euiLink--primary"
                        href="/app/home#/tutorial_directory"
                        rel="noreferrer"
                      >
                        add sample data 
                      </a>
                    </EuiLink>
                    to OpenSearch Dashboards.
                  </p>
                </div>
              </EuiText>
            </span>
          </EuiTextColor>
        </div>
      </EuiEmptyPrompt>
    </div>
  </EuiPageBody>
</CreateIndex>
`;
